### PR TITLE
#461 Prohibit a semicolon in the end of try-with-resourses head

### PR DESCRIPTION
**Pull request by HDouss**
Resolving #461 Prohibit a semicolon in the end of try-with-resourses head.
Checkstyle consider final (optional) semicolon as belonging to the RESOURCE_SPECIFICATION token, so it is enough to test if RESOURCE_SPECIFICATION contains a semicolon.

### CODE REVIEW

**Comment by HDouss** [comment]
@davvd PR is ready for review

**Comment by davvd** [comment]
@hdouss Thanks, someone will review your pull request soon

**Comment by davvd** [comment]
@dmzaytsev review this one, please

File: qulice-checkstyle/src/main/java/com/qulice/checkstyle/FinalSemicolonInTryWithResourcesCheck.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import com.puppycrawl.tools.checkstyle.api.Check;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Checks that try-with-resources does not end with a semicolon.
+ *
+ * @author Hamdi Douss (douss.hamdi@gmail.com)
+ * @version $Id$
+ */
```
Comment (dmzaytsev) [inline qulice-checkstyle/src/main/java/com/qulice/checkstyle/FinalSemicolonInTryWithResourcesCheck.java]: @HDouss since 0.14.2 I guess

**Comment by dmzaytsev** [comment]
@HDouss will it work on valid code like this
```
 try (
           final Closeable door = new Door();
            final Closeable window = new Window();
            final Closeable win = new Window()
        ) {
```
 I think it would be good to add one more tests
so one test for  invalid code and one test for valid code
what do you think?

**Comment by HDouss** [comment]
@dmzaytsev 
>I think it would be good to add one more tests
so one test for invalid code and one test for valid code
what do you think?

I agree.

**Comment by HDouss** [comment]
@dmzaytsev 
>@HDouss since 0.14.2 I guess

We never added the since tag before. Do you suggest adding it for `FinalSemicolonInTryWithResourcesCheck` class ?

**Comment by dmzaytsev** [comment]
@krzyk what do you think ?

I see the since tag everywhere, so I thought it is the requirement :)

> We never added the since tag before. Do you suggest adding it for FinalSemicolonInTryWithResourcesCheck class ?

**Comment by HDouss** [comment]
@dmzaytsev At least, none of our custom checkstyle or PMD rule classes have the `@since` tag

**Comment by HDouss** [comment]
@dmzaytsev

>I think it would be good to add one more tests
so one test for invalid code and one test for valid code
what do you think?

Done !

**Comment by dmzaytsev** [comment]
@HDouss thank you

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by davvd** [comment]
> @davvd PR is ready for review

@hdouss thank you, I will check it

**Comment by krzyk** [comment]
@HDouss for now let's leave `since` tag out

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by krzyk** [comment]
@rultor deploy pls

**Comment by davvd** [comment]
@dmzaytsev Much appreciated! **20 mins** was added to your account, payment ID `73452962`, time spent is 24 hours and 31 mins

review comments (c=5) added as a bonus

+20 added to your rating, at the moment it is: [+1254](http://www.netbout.com/b/35166?open=rating)

**Comment by davvd** [comment]
@rultor deploy now pls

**Pull request merged**